lib/MusicBrainz/Server/Data/Editor.pm

```diff
@@ -186,9 +186,16 @@ sub find_by_ip {
 sub search_by_email {
     my ($self, $email) = @_;
 
+    # Remove periods (\.), and anything preceded by a + sign on the user side
+    # for email matching, since both of these are often used as "free aliases"
+    # by email providers
+    $email =~ s/\\\+.*@/@/;
+    $email =~ s/\\\.(.*@)/$1/g;
```

Comment:
Since `$email` can be any POSIX regexp, let’s not assume it necessarily contains a `@` symbol:

```suggestion
    $email =~ s/\\\+[^@]*//;
    $email =~ s/\\\.([^@]*)/$1/g;
```

Reply:
I'm not sure we necessarily want to remove everything after a `+` if there's no `@`, I'd leave that as-is. Removing periods everywhere is probably kinda-sorta fine, but I'm also not sure - what happens if we search for a specific provider (that is known for spam) for example? It wouldn't match since we don't remove the periods from the provider on the PSQL, right?

Reply:
Ok, but then we should explain that searching for an email by user info should be followed with an `@`. For example: `user.info+tag@example.org`can be found with either `user\.info\+tag@` or `userinfo@` but not `user\.info\+tag`.

Reply:
Explanation added through https://github.com/metabrainz/musicbrainz-server/pull/2074/commits/f6228ab9ebddeaf5ab703b81c6580b714bec447a